src/components/SplitVertical/SplitVertical.jsx

```diff
@@ -0,0 +1,392 @@
+		onResize: func,
+		/**
+		 * Use this prop to shift the collapsed position by a known value.
+		 */
+		collapseShift: number,
+	},
+
+	components: {
+		/**
+		 * Left pane of the split.
+		 */
+		LeftPane: createClass({
+			displayName: 'SplitVertical.LeftPane',
+			propTypes: {
+				/**
+				 * Any valid React children.
+				 */
+				children: node,
+				/**
+				 * Set width of this pane.
+				 */
+				width: oneOfType([number, string]),
+				/**
+				 * Define this pane as the primary content pane. When the split is collapsed, this pane becomes full width.
+				 */
+				isPrimary: bool,
+			},
+			getDefaultProps(){
+				return {
+					isPrimary: false,
+				};
+			},
+		}),
+
+		/**
+		 * Right pane of the split.
+		 */
+		RightPane: createClass({
+			displayName: 'SplitVertical.RightPane',
+			propTypes: {
+				/**
+				 * Any valid React children.
+				 */
+				children: node,
+				/**
+				 * Set width of this pane.
+				 */
+				width: oneOfType([number, string]),
+				/**
+				 * Define this pane as the primary content pane. When the split is collapsed, this pane becomes full width.
+				 */
+				isPrimary: bool,
+			},
+			getDefaultProps(){
+				return {
+					isPrimary: false,
+				};
+			},
+		}),
+
+		/**
+		 * The area that separates the split panes. Can be dragged to resize them.
+		 */
+		Divider: createClass({
+			displayName: 'SplitVertical.Divider',
+			propTypes: {
+				/**
+				 * Any valid React children.
+				 */
+				children: node,
+			},
+		}),
+	},
+
+	getDefaultProps() {
+		return {
+			isExpanded: true,
+			isAnimated: false,
+			collapseShift: 0,
+			onResize: _.noop,
+		};
+	},
+
+	getInitialState() {
+		return {
+			isAnimated: false,
+		};
+	},
+
+	getPanes() {
+		const { children } = this.props;
+		const {
+			leftPane: leftPaneRef,
+			rightPane: rightPaneRef,
+		} = this.refs;
+
+		const leftPaneElement = _.get(filterTypes(children, SplitVertical.LeftPane), 0, <SplitVertical.LeftPane />)
+		const rightPaneElement = _.get(filterTypes(children, SplitVertical.RightPane), 0, <SplitVertical.RightPane />)
+		let primaryElement, primaryRef;
+		let secondaryElement, secondaryRef;
+
+		if (leftPaneElement.props.isPrimary && !rightPaneElement.props.isPrimary) {
+			primaryElement = leftPaneElement;
+			primaryRef = leftPaneRef;
+			secondaryElement = rightPaneElement;
+			secondaryRef = rightPaneRef;
+		} else {
+			primaryElement = rightPaneElement;
+			primaryRef = rightPaneRef;
+			secondaryElement = leftPaneElement;
+			secondaryRef = leftPaneRef;
+		}
+
+		return {
+			left: leftPaneElement.props,
+			right: rightPaneElement.props,
+			primary: primaryElement.props,
+			primaryRef,
+			secondary: secondaryElement.props,
+			secondaryRef,
+		};
+	},
+
+	applyDeltaToSecondaryWidth(dX, isExpanded, secondaryStartRect, secondaryRef, secondary, right, innerRef, primaryRef, collapseShift=0) {
+		if (isExpanded) {
+			secondaryRef.style.flexBasis = `${secondaryStartRect.width + dX * (secondary === right ? -1 : 1)}px`;
```

Comment:
with so many dom elements and logic paths in this function, it was easier to have it mutate the dom nodes directly rather than doing the mutation externally. (have to mutate the dom nodes here for perf reasons to shortcut the state -> rerender cycle)